python/ray/experimental/locations.py

```diff
@@ -35,6 +36,16 @@ def get_object_locations(
     """
     if not ray.is_initialized():
         raise RuntimeError("Ray hasn't been initialized.")
+    # This call can be expensive if not called by the driver.
+    # If this is for purely metrics use, we can skip the call for performance.
+    if (
+        skip_get_locations
+        and not ray.data.context.DataContext.get_current().enable_get_object_locations_for_metrics
```

Comment:
this function is supposed to be used by all Ray libs, not only Data. So let's not add data-specific logic here. 
I guess we don't need to touch this function at all. We can just skip calling it in Data code.